src/chi/javascript/components/sidenav.js

```diff
@@ -96,6 +98,46 @@ class Sidenav extends Component {
     });
   }
 
+  singleLevelMenuItems() {
+    this._elem
+    .querySelectorAll(`nav ul.${LINKLIST_CLASS} li`)
+    .forEach(singleLevelMenuItem => {
+      const menuElementLink = singleLevelMenuItem.querySelector('a').getAttribute('href');
+
+      if (!this._elem.querySelector(`.chi-drawer${menuElementLink}`)) {
+        let menuItemToActivate;
+
+        singleLevelMenuItem.addEventListener("click", (e) => {
+          const activeMenuItem = this.getActiveMenuItem();
+
+          for (
+            let cur = e.target;
+            cur && !cur.classList.contains(LINKLIST_CLASS);
+            cur = cur.parentNode
+          ) {
+            menuItemToActivate = cur;
```

Comment:
Wouldn't it better to check if cur is the element you are looking for? So then, you'll be able check if it existed, and if not, alert the developer that he didn't build the component properly.